ocaml/idl/ocaml_backend/context.ml

```diff
@@ -155,6 +159,55 @@ let trackid_of_session ?(with_brackets=false) ?(prefix="") session_id =
 let trackid ?(with_brackets=false) ?(prefix="") __context = (* CP-982: create tracking id in log files to link username to actions *)
   trackid_of_session ~with_brackets ~prefix __context.session_id
 
+let _clean_addr_of_string ip =
+  (* in the IPv4 case, users should see 127.0.0.1 rather than ::ffff:127.0.0.1 *)
+  let ipv4_affix = "::ffff:" in
+  ( if Astring.String.is_prefix ~affix:ipv4_affix ip then
+      Astring.String.drop ~max:(String.length ipv4_affix) ip
+  else
+    ip
+  )
+  |> Ipaddr.of_string
```

Comment:
This needs to be converted to `Some _`, as it was in the original commit, otherwise the type checker will complain in `_client_of_rq`

Reply:
at some point the type of this function changed, but it now outputs an option :) 

https://github.com/mirage/ocaml-ipaddr/blob/7057270eda22ce19f56478f5ed7479ac9a6d0a9c/lib/ipaddr.mli#L54

Reply:
ah, that's confusing